students/javarb/sessions/2019/may/Session_34-35-Serialization/Session_34-35-Serialization.md

```diff
@@ -0,0 +1,225 @@
+
+```java
+public class Serialization {
+    public static void main(String[] args) {
+        new JacksonBenchmark().benchmark(150, 200000);
+        new GsonBenchmark().benchmark(150, 200000);
+        new FastjsonBenchmark().benchmark(150, 200000);
+    }
+}
+```
+
+Each type of bechmark inherit from abstract class `Benchmark`. In that class,  `bechmark()` contructor  method is implemented, altogether with `mean()` and `runBenchmark()`, the latter is defined as an abstract method that is implemented in each subclass:
+
+```java
+public abstract class Benchmark {
+    public void benchmark(int samplesize, int objects) {
+        List<Person> persons = new ArrayList<>();
+
+        for (int i = 0; i < objects; i++) {
+            Person person = new Person();
+            person.setId(i);
+            person.setName("Name_" + i);
+            person.setAge(i % 100);
+            persons.add(i, person);
+        }
+
+        List<Long> results = new ArrayList<>();
+
+        for (int i = 0; i < samplesize; i++) {
+            results.add(runBenchmark(persons));
+        }
+
+        long mean = mean(results);
+        System.out.println("Result of " + getClass().getSimpleName() + " = "+ mean + " ms");
+
+    }
+
+    protected abstract Long runBenchmark(List<Person> persons);
+
+    private long mean(List<Long> benchmarks) {
+        long sum = 0;       
+        for (Long benchmark : benchmarks) {
+            sum += benchmark;
+        }
+        return sum / benchmarks.size();
+    }
+}
+```
+
+Then, when a specific implementation is running the constructor, for example for, for Jackson `new JacksonBenchmark().benchmark(150, 200000);` this will execute `runBenchmark()` for that specific library implementation, in this case called `JacksonBenchmark` which is inheriting from `Benchmark` class and implementing `runBenchmark()` method according to library specifications:
+
+```java
+public class JacksonBenchmark extends Benchmark {
+    @Override
+    protected Long runBenchmark(List<Person> persons) {
+        long begin = System.currentTimeMillis();
+        try {
+            new ObjectMapper().writeValueAsString(persons);
+        } catch (JsonProcessingException e) {
+            e.printStackTrace();
+        }
+        return System.currentTimeMillis() - begin;
+    }
+}
+```
+**Note:** `@Override` annotation specifies we are overriding the specfied method, this is specially useful for a [couple of reasons](https://stackoverflow.com/a/94411/4278635):
+- Compiler will warn us if there is a misspelling in the name of our method, or wrong use of method's parameters.
+- Make code easier to understand because becomes obvious what that method is doing.
+
+Other implementations can be seen on correspondig code project on this same directory. 
+
+#### Part 3 - Blog 
+
+I opened a blog on wordpress to keep this and another technology things https://jaarit.wordpress.com/ 
```

Comment:
Oh nice! I just thought a markdown file would do, as long as it was written as a blog post. We could then generate a github-pages site for this repo where we could showcase your (yours, and the others') blog post. But if you prefer to have your own blog I certainly applaud that!

Reply:
I think that if you are on agree, we could handle both feed internat github-pages showcase and also I could publish material of mine on my wordpress site. I like that idea :)